src/Autosuggest.js

```diff
@@ -238,13 +262,17 @@ class Autosuggest extends Component {
             closeSuggestions('enter');
 
             if (focusedSuggestion !== null) {
+              const newValue = getSuggestionValue(focusedSuggestion);
+
               onSuggestionSelected(event, {
                 suggestion: focusedSuggestion,
-                suggestionValue: value,
+                suggestionValue: newValue,
                 sectionIndex: focusedSectionIndex,
                 method: 'enter'
               });
-              this.maybeCallOnSuggestionsUpdateRequested({ value, reason: 'enter' });
+
+              this.maybeCallOnChange(event, newValue, 'enter');
```

Comment:
Let's add `'enter'` to the possible `method` values [in the docs](https://github.com/moroshko/react-autosuggest#inputprops-required):

    'enter' - user pressed Enter after hovering with mouse over a suggestion